gapic/templates/%namespace/%name_%version/%sub/services/%service/client.py.j2

```diff
@@ -480,6 +480,19 @@ class {{ service.client_name }}(metaclass={{ service.client_name }}Meta):
         {{ "\n" }}
     {% endfor %}
 
+    def __enter__(self):
+        self.transport.__enter__()
+        return self
```

Comment:
I'm not sure why we expect `transport` to behave with context manager semantics, but yet cannot use them so.  Don't we really just want a `close` method on the transport, to be called in our `__exit__`?

Reply:
What do you mean by "but yet cannot use them so"? Nothing stops developers from using transport as context manager. 

I am unsure if we want `transport` to be used a context manager. Intuitively, it does not need to be. As long, each `transport` knows how to `close` we would get the desired functionality (i.e. client = context manager) and also perhaps have a cleaner interface. 

@software-dov any thoughts?

Reply:
Having one context manager call another object's `__enter__` and `__exit__` does not feel like idiomatic Python usage:  as with other "dunder" methods, such "protocol helpers" are not intended to be called directly.

Reply:
I was also wondering about this, but after reading some posts on [SO](https://stackoverflow.com/questions/26635684/calling-enter-and-exit-manually), I came to conclusion it's not a bad thing.

Reply:
Thank you for your suggestions. I'm going to replace `__enter__`, `__exit__` in `transport` with `close`. I'll leave `close` from `client` at this moment, but we can consider adding it in the future. 

@tseaver @tswast @software-dov @busunkim96